csrc/scheduler/vectorize_helper.cpp

```diff
@@ -484,6 +484,29 @@ ContiguousInnerDimensionsMapper::computeInfoC2P(
           from->getRootDomain().begin(),
           from->getRootDomain().begin() + clear_pos + 1);
     }
+
+    // A special case is that, if the last broadcast dimension resolved in
+    // consumers root domain is mapped for vectorization, then merge order in
+    // the vectorization axes matters. T0[i0, i1] T1[i0, i1, b2] = broadcast(T0)
+    // T2[i0, i1, i3]
+    // T3[i0, i1, i2] = T1 + T2
+    //
+    // If the mapped ids are {i0, i2, i1}, when propagating from T3 to T1, the
+    // previous "inner most resolved broadcast" would eliminate both i0 & i1
+    // from vectorization, but the reality is that the merged axes has i1 in the
+    // inner most and we should still map it for vectorization. see issue:
+    // https://github.com/NVIDIA/Fuser/issues/1567#issuecomment-1894605385
+    bool clear_pos_in_mapped_root = false;
+    for (auto i : c10::irange(from_ids.size())) {
+      if (clear_pos_in_mapped_root) {
+        if (auto iter = consumer_ids_to_clear.find(from_ids[i]);
+            iter != consumer_ids_to_clear.end()) {
+          consumer_ids_to_clear.erase(iter);
+        }
+      } else if (from_ids[i] == from->getRootDomain()[clear_pos]) {
```

Comment:
Shouldn't we have a guard like `clear_pos >= 0`?

Reply:
good catch! I should have kept it inside the true block for `clear_pos >= 0`

I'll try to add a test case for that!

Reply:
put it under the block. I did verify the invalid access on a simple repro but don't see a good reason to put a test.
The existing code didn't break any test, looks like that `-1` accidentally just give us a false on this `from_ids[i] == from->getRootDomain()[clear_pos]`...